zipline/modelling/term.py

```diff
@@ -0,0 +1,294 @@
+"""
+Base class for Filters, Factors and Classifiers
+"""
+from numpy import (
+    empty,
+    float64,
+    full,
+    nan,
+)
+
+from zipline.errors import (
+    InputTermNotAtomic,
+    TermInputsNotSpecified,
+    WindowLengthNotPositive,
+    WindowLengthNotSpecified,
+)
+from zipline.utils.lazyval import lazyval
+
+
+NotSpecified = (object(),)
+
+
+class Term(object):
+    """
+    Base class for terms in an FFC API compute graph.
+    """
+    inputs = NotSpecified
+    window_length = NotSpecified
+    domain = None
+    dtype = float64
+
+    _term_cache = {}
```

Comment:
do we care that the `Term` object will never be gc'd? should this maybe be a `WeakValueDict`?